code/game/machinery/Sleeper.dm

```diff
@@ -81,8 +82,19 @@
 	go_out()
 
 /obj/machinery/sleeper/process()
-	if(filtering > 0)
-		if(beaker)
+	for(var/mob/M as mob in src) // makes sure that simple mobs don't get stuck inside a sleeper when they resist out of occupant's grasp
+		if(M == occupant)
+			continue
+		else
+			M.forceMove(loc)
+
+	if(occupant)
+		if(autoeject_dead && occupant.is_dead())
+			playsound(src.loc, 'sound/machines/buzz-sigh.ogg', 40)
+			go_out()
+			return
+
+		else if((filtering > 0) && beaker)
 			// To prevent runtimes from drawing blood from runtime, and to prevent getting IPC blood.
 			if(!istype(occupant) || !occupant.dna || (NO_BLOOD in occupant.dna.species.species_traits))
```

Comment:
You've lost this feature now. If an occupant leaves the sleeper that filtering will stop on it's own

Reply:
I'm not sure what you mean by this.

Reply:
In current code the filtering option is turned off when an occupant leaves the sleeper. In your PR this wont happen since this code is only executed when there is an occupant

Reply:
Ah I understand now, however we have this in the go_out proc:
```js
/obj/machinery/sleeper/proc/go_out()
	if(filtering)
		toggle_filter()
```
This ensures filtering is toggled when they exit.

Reply:
yeah okay that should cover it all then. Thanks for the clarification!